[wip] Add requests as a dependency

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25153 [wip] Add requests as a dependency**
* #25152 [testing][not for land] everything is important

Trying to special-case this for tests only was getting way too complicated so let's just add `requests` as a dependency flat-out

Commit:
[wip] Install requests for binary tests

[ci binary_linux_conda_2.7_cpu_devtoolset7_build]
[ci binary_linux_conda_2.7_cpu_devtoolset7_test]
[ci binary_macos_conda_2.7_cpu_build]
[ci binary_macos_conda_2.7_cpu_test]

Commit:
Update on "[wip] Install requests for binary tests"


[ci binary_linux_conda_2.7_cpu_devtoolset7_build]
[ci binary_linux_conda_2.7_cpu_devtoolset7_test]
[ci binary_macos_conda_2.7_cpu_build]
[ci binary_macos_conda_2.7_cpu_test]

Commit:
Update on "[wip] Install requests for binary tests"


[ci binary_linux_conda_2.7_cpu_devtoolset7_build]
[ci binary_linux_conda_2.7_cpu_devtoolset7_test]
[ci binary_macos_conda_2.7_cpu_build]
[ci binary_macos_conda_2.7_cpu_test]

Commit:
Update on "[wip] Install requests for binary tests"


[ci binary_linux_conda_2.7_cpu_devtoolset7_build]
[ci binary_linux_conda_2.7_cpu_devtoolset7_test]
[ci binary_macos_conda_2.7_cpu_build]
[ci binary_macos_conda_2.7_cpu_test]

zrphercule reviewed (approved):
Please make sure to merge this fix asap (after fix all the bugs)

Commit:
Update on "[wip] Add requests as a dependency"


Trying to special-case this for tests only was getting way too complicated so let's just add `requests` as a dependency flat-out

Commit:
Update on "[wip] Add requests as a dependency"


Trying to special-case this for tests only was getting way too complicated so let's just add `requests` as a dependency flat-out
